File: JAVA/2283. Check if Number Has Equal Digit Count and Digit Value.java
class Solution {
    public boolean digitCount(String n) {
        int[] c=new int[10];
        for(int i=0;i<n.length();i++){
            c[n.charAt(i)-'0']++;
        }
        for(int i=0;i<n.length();i++){
            if((n.charAt(i)-'0')!=c[i])
            return false;
        }
        return true;
    }
}

File: C++/2283. Check if Number Has Equal Digit Count and Digit Value.cpp
class Solution {
public:
    bool digitCount(string n) {
        map<int,int>m;
        for(char c:n){
            m[(c-'0')]++;
        }
        for(int i=0;i<n.size();i++){
            if(m[i]!=(n[i]-'0'))
            return false;
        }
        return true;
    }
};